### PR TITLE
Add documentation to examples

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is
up to date with these changes.

### CODE REVIEW

File: examples/cfd/ahmed_body_mgn/README.md [hunk 1]
```diff
@@ -11,6 +11,12 @@ the drag coefficient.
 The dataset for this example is not publicly available. To get access, please reach out
 to the NVIDIA Modulus team at <simnet-team@nvidia.com>.
```
[the remark this answers is not in the document]
Comment (ram-cherukuri) [reply examples/cfd/ahmed_body_mgn/README.md:12]: Also can we spell out what optional geometry parameters are?

File: examples/weather/fcn_afno/README.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+Neural Operator (AFNO) attention](https://openreview.net/pdf?id=EXHG-A3jlM)
+
+![Comparison between the FourCastNet and the ground truth (ERA5) for $u-10$ for
+different lead times.](../../../docs/img/FourCastNet.gif)
+
+## Dataset
+
+The model is trained on a subset of the ERA5 reanalysis data on single levels and
+pressure levels that is pre-processed and stored into HDF5 files.
+The subset of the ERA5 training data that FCN was trained on is hosted at the
```
Comment (ktangsali) [inline examples/weather/fcn_afno/README.md:40]: Did you test this link? For some reason I can only view the files but the download option is deactivated for me.

File: examples/cfd/ahmed_body_mgn/README.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+address a wide variation in Reynolds number. We identify the design points using the
+Latin hypercube sampling scheme for space filling design of experiments and generate
+around 500 design points.
+
+The aerodynamic simulations were performed using the GPU-accelerated OpenFOAM solver
+for steady-state analysis, applying the SST K-omega turbulence model. These simulations
+consist of 7.2 million mesh points on average, but we use the surface mesh as the input
+to training which is roughly around 70k mesh nodes.
+
+To request access to the full dataset, please reach out to the
```
Comment (ktangsali) [inline examples/cfd/ahmed_body_mgn/README.md:34]: Are we planning to release a subset of data via NGC for easy access?

File: examples/weather/fcn_afno/README.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+Data parallelism is also supported with multi-GPU runs. To launch a multi-GPU training,
+run
+
+```bash
+mpirun -np <num_GPUs> python train_era5.py
+```
+
+If running inside a docker container, you may need to include the `--allow-run-as-root`
+in the multi-GPU run command.
+
```
Comment (ktangsali) [inline examples/weather/fcn_afno/README.md:78]: At some point we were thinking of adding a minimal inference script to each example that shows how to run the model in inference mode. Do we want to do that in a separate PR or should we add it in this one itself?

**Comment by ktangsali** [comment]
Based on today's discussion, should the target branch be the rc branch and not `main` ?

File: examples/cfd/ahmed_body_mgn/README.md (hunk 3 above)
Comment (ram-cherukuri) [reply examples/cfd/ahmed_body_mgn/README.md:34]: For now, lets have the model package on NGC with the checkpoint, inference script and a few sample geometries and corresponding ground truth included.

File: examples/weather/fcn_afno/README.md (hunk 4 above)
Comment (ram-cherukuri) [reply examples/weather/fcn_afno/README.md:78]: Lets keep it seperate for the following reason - our inference template is to provide the model package with a checkpoint.

**Comment by NickGeneva** [comment]
Yeah, merging to RC branch would be to get this documentation in the tagged commit. Not the biggest deal but I suppose this would be nice.

File: examples/weather/fcn_afno/README.md (hunk 4 above)
Comment (mnabian) [reply examples/weather/fcn_afno/README.md:78]: I agree with Ram. We should also have a unified inference recipe for our weather prediction models.

File: examples/cfd/ahmed_body_mgn/README.md (hunk 3 above)
Comment (mnabian) [reply examples/cfd/ahmed_body_mgn/README.md:34]: Let us track this on a separate issue/PR: https://github.com/NVIDIA/modulus-launch/issues/78

**Review by ram-cherukuri — COMMENTED** [review]
Looks good to me.

File: examples/weather/graphcast/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+## Problem overview
+
+GraphCast is a multi-scale graph neural network-based autoregressive model. It is
+trained on historical weather data from ECMWF's ERA5 reanalysis archive. GraphCast
+generates predictions at 6-hour time steps for a set of surface and atmospheric
+variables. This prediction covers a 0.25-degree latitude-longitude grid,
+providing approximately 25 x 25 kilometer resolution at the equator.
+
+## Dataset
+
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/weather/graphcast/README.md:15]: As we discussed, the CDS data needs to be processed and the scripts are not ready yet. Will stick to the NERSC dataset for now for SFNO, AFNO, and GraphCast

File: examples/weather/fcn_sfno/README.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+large-ensemble forecasts with thousands of ensemble-members for improving probabilistic
+forecasting. We discuss how data-driven deep learning models such as FourCastNet are a
+valuable addition to the meteorology toolkit to aid and augment NWP models.
+
+FourCastNet is based on the [vision transformer architecture with Adaptive Fourier
+Neural Operator (AFNO) attention](https://openreview.net/pdf?id=EXHG-A3jlM)
+
+## Dataset
+
+We rely on DeepMind's vortex shedding dataset for this example. The dataset includes
```
[the remark this answers is not in the document]
Comment (mnabian) [reply examples/weather/fcn_sfno/README.md]: Hmm this is an outdated version

**Comment by ktangsali** [comment]
Looks good to me!

**Review by ktangsali — APPROVED** [review]
